### DISCUSSION TITLE
Stack FluxPoints (v0.20/1.0)

### DISCUSSION THREAD

**Discussion by Elisa-Visentin** [gammapy/gammapy#4756] [Help] [answered]
Hi, I'm facing some troubles while trying to stack (i.e., merge) two different FluxPoints objects (from two different experiments). The number of their columns and the Energy units are different due to the different data format and software used (1 satellite, 1 Cherenkov telescope). One of them must be read from a .fits file that contains flux values, the other one must be produced by GammaPy from observations.
![image](https://github.com/gammapy/gammapy/assets/121040436/2ce9bac8-254e-4170-96ed-51ddefce4c55)
![image](https://github.com/gammapy/gammapy/assets/121040436/9498d1d2-d5a8-43c2-b7f1-960d7622a3ae)
`abc=FluxPoints.from_stack([flx,flux_points], axis=1)`
```
File /fefs/home/elisa.visentin/mambaforge-pypy3/envs/gammapy-v0.20.1/lib/python3.8/site-packages/gammapy/estimators/map/core.py:805, in FluxMaps.from_stack(cls, maps, axis, meta)
    802 data = {}
    804 for quantity in reference.available_quantities:
--> 805     data[quantity] = Map.from_stack(
    806         [_._data[quantity] for _ in maps], axis=axis
    807     )
    809 if meta is None:
    810     meta = reference.meta.copy()

File /fefs/home/elisa.visentin/mambaforge-pypy3/envs/gammapy-v0.20.1/lib/python3.8/site-packages/gammapy/maps/core.py:1538, in Map.from_stack(cls, maps, axis, axis_name)
   1535         m_geom = m.geom
   1537     if not m_geom == geom:
-> 1538         raise ValueError(f"Image geometries not aligned: {m.geom} and {geom}")
   1540     data.append(m.quantity.to_value(maps[0].unit))
   1542 return cls.from_geom(
   1543     data=np.stack(data), geom=geom.to_cube(axes=[axis]), unit=maps[0].unit
   1544 )

ValueError: Image geometries not aligned: RegionGeom

	region     : NoneType
	axes       : ['lon', 'lat', 'energy']
	shape      : (1, 1, 12)
	ndim       : 3
	frame      : icrs
	center     : nan deg, nan deg
 and RegionGeom

	region     : NoneType
	axes       : ['lon', 'lat', 'energy']
	shape      : (1, 1, 15)
	ndim       : 3
	frame      : icrs
	center     : nan deg, nan deg


```
Can I stack them? How? (Sorry, I'm a GammaPy beginner)

Thank you

**Comment by registerrier** [comment]
Hi @Elisa-Visentin , you cannot stack/merge `FluxPoints` objects. Maybe you could clarify what you want to obtain here?

The `FluxPoints.from_stack()` constructor is meant to build a higher dimensionality object (such as lightcurves or flux profiles) from a list of aligned `FluxPoints` (i.e. same energy binning). 

Gammapy requires that flux points objects have adjacent energy bins so that flux points from different experiments reside in different `FluxPoint` objects. If you want to perform a fit on two different SEDs you can always create two `FluxPointsDataset`.

**Reply by Elisa-Visentin** [reply]
Hi @registerrier, sorry, I misunderstood the Doc, my fault. I have two FluxPoints (15 points (100 MeV-2 TeV) the former, 12 (50 GeV-20 TeV) the latter) and I can fit them individually (so, I evaluate two curves, one per energy range). I would like to 'merge' the two datasets into a single one to fit the 27 points (15+12) by a single curve that will span a greater energy range (100 MeV-20 TeV) and fit the data collected by both experiments.

**Reply by registerrier** [reply, marked as answer]
If you fit them with a `FluxPointsDataset` you can create one for each SED and fit them jointly. This is what is done in the following tutorial:
https://docs.gammapy.org/1.0.1/tutorials/analysis-1d/sed_fitting.html#sphx-glr-tutorials-analysis-1d-sed-fitting-py

Conversely, if you reduce the VHE dataset with gammapy you can even jointly fit the SED on one hand and the VHE dataset on the other. This is what is done here:
https://docs.gammapy.org/1.0.1/tutorials/analysis-3d/analysis_mwl.html#sphx-glr-tutorials-analysis-3d-analysis-mwl-py

Does it help?

**Reply by Elisa-Visentin** [reply]
Thank you really much, I need the first one.
Could I suggest a Doc upgrade? `Create flux points by stacking list of flux points` is a bit confusing (I intended it as pandas vstack).
Anyway, thank you really much!
